materialchart/src/main/java/app/priceguard/materialchart/Chart.kt

```diff
@@ -110,8 +111,28 @@ class Chart @JvmOverloads constructor(
 
     override fun onDraw(canvas: Canvas) {
         super.onDraw(canvas)
-        drawXAxis(canvas, xAxisPaint)
-        drawYAxis(canvas, yAxisPaint)
+        dataset ?: return
+        if (dataset?.showXAxis == false) {
+            if (dataset?.showYAxis == false) {
+                xAxisMargin = zeroDp
+                xGraphPadding = zeroDp
+            } else {
+                yAxisMargin = zeroDp
+                drawYAxis(canvas, yAxisPaint)
+            }
+        }
+        else if (dataset?.showYAxis == false) {
+            if (dataset?.showXAxis == false) {
+                yAxisMargin = zeroDp
+                yGraphPadding = zeroDp
```

Comment:
I think `if (dataset?.showXAxis == false) {}` can be removed because showAxis is always true in `else if (dataset?.showYAxis == false) {}`

Reply:
I use that part because of code consistency.
But now, I understand this code doesn't work what I expected, so I change conditions.